### PR TITLE
CV2-3555: add a bulk mutation to make items read/unread

### PR DESCRIPTION
**Pull request by melsawy**
## Description

Add a bulk mutation to make items read/unread.

References: CV2-3555

## How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can verify the changes. Please describe whether or not you implemented automated tests.

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: app/graph/mutations/bulk_mark_read_mutation.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module Mutations
+  class BulkMarkReadMutation < BaseMutation
```
[the remark this answers is not in the document]
Comment (caiosba) [reply app/graph/mutations/bulk_mark_read_mutation.rb:2]: Don't worry about it.

**Review by caiosba — COMMENTED** [review]
I'll leave this one for @hartsick (plz) to review tomorrow.

**Comment by hartsick** [comment]
based on what I understand of this PR, the GraphQL side of things looks great to me. nice work, @melsawy!

I just have one question - do we expect to use the BulkMarkReadMutation base class for other resources in the future? or is this the only time? the reason I ask is that it's different conceptually than the base classes we already have (update, create, and destroy) in that they don't assume any specific fields in the database. mark read does, since it requires `read` to be an attribute on the model the mutation is declared on.

because of that coupling, I wonder if BulkMarkReadMutation shouldn't be a base class at all, and instead ProjectMediaMutations::Bulk::MarkRead should just be implemented entirely in ProjectMediaMutations. something like this (just collapsing the code you've already written into one location - I haven't tried this locally, so some other work may be needed to get it running):

```
module ProjectMediaMutations
  ...
  class BulkMarkRead < BaseMutation
    class << self
      def inherited(subclass)
        parents_mapping = GraphqlCrudOperations.hashify_parent_types(subclass.module_parent::PARENTS)
        mutation_target = subclass.module_parent.module_parent::MUTATION_TARGET

        define_shared_bulk_behavior(:mark_read, subclass, mutation_target, parents_mapping)

        subclass.field :updated_objects, ["#{mutation_target.camelize}Type".constantize, null: true], null: true, camelize: false
      end
    end
    
     description "Allow multiple items to be marked as read or unread."

      graphql_name "BulkProjectMediaMarkRead"

      argument :read, GraphQL::Types::Boolean, "A boolean value for ProjectMedia read value", required: true, camelize: false
  end
end
```

**Comment by caiosba** [comment]
Good point @hartsick ! I also agree that probably that mutation shouldn't be a base one, but @melsawy can confirm.

**Pull request merged**